### PR TITLE
docs: add dev server function config API

### PR DESCRIPTION
**Pull request by lmiller1990**
Docs for https://github.com/cypress-io/cypress/pull/23605

### CODE REVIEW

**Review by astone123 — APPROVED** [review]
Looks good

**Comment by lmiller1990** [comment]
Just waiting on original feature to be review by Brian.

**Comment by brian-mann** [comment]
From looking at the implementation code it seems like `baseConfig` is yielded to the async `webpackConfig` function, but no example demonstrates that

**Pull request merged**
